src/error.ts

```diff
@@ -208,4 +208,13 @@ export class DBOSConflictingWorkflowError extends DBOSError {
   constructor(workflowID: string, msg: string) {
     super(`Conflicting workflow invocation with the same ID (${workflowID}): ${msg}`, ConflictingWorkflowError);
   }
-}
\ No newline at end of file
+}
+
+const MaximumRetriesError = 23;
+export class DBOSMaxStepRetriesError extends DBOSError {
+  readonly errors;
+  constructor(stepName: string, maxRetries: number, errors: Error[]) {
+    super(`Step ${stepName} has exceeded its maximum of ${maxRetries} retries. Previous errors: ${errors.map(e => e.message).join(', ')}`, MaximumRetriesError);
```

Comment:
Do you think we should label the error by the number of retry? Such as `Error 1: `, `Error 2: `.... Otherwise, a long list could be hard to parse.

Reply:
Done!